src/main/java/algorithms/leetcode/easy/BackSpaceStringCompare.java

```diff
@@ -0,0 +1,21 @@
+package algorithms.leetcode.easy;
+
+public class BackSpaceStringCompare {
+    public static boolean backspaceCompare(String s, String t) {
+        String string1 = checkAfterBackspacing(s);
+        String string2 = checkAfterBackspacing(t);
```

Comment:
`string1` and `string2` are getting used only once. so avoid these variables and directly call functions `checkAfterBackspacing(s).equals(checkAfterBackspacing(t))`